dataplicity/m2mmanager.py

```diff
@@ -135,37 +136,45 @@ def get_terminal(self, name):
     def on_instruction(self, sender, data):
         """Instructions sent by privileged client."""
         log.debug("instruction %r from %s", data, sender)
-        action = data['action']
-        if action == 'sync':
+
```

Comment:
so this function call is executed after the binary string is decoded?  Maybe the .decode("utf-8") could be located in the bencode.py then?

Reply:
The majority of the time, it needs to be binary data. Alas, bencode has no way to distinguish plain bytes from unicode strings.